games/guess-it/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+extern crate unescape;
+use std::fmt;
+use std::fmt::Formatter;
+use scrypto::prelude::*;
+use sbor::{Encode,Decode,TypeId, Describe};
+use serde::{Serialize, Deserialize};
+use unescape::unescape;
+use std::cmp;
+mod context;
+
+#[derive(Debug, Copy, Clone, TypeId, Encode, Decode, Describe, PartialEq, Eq, Deserialize, Serialize)]
+#[repr(u32)]
+pub enum State {
+    AcceptingChallenger = 0,
+    MakeGuess = 1,
+    SubmitSecret = 2,
+    WinnerSelection = 3,
+    Payout = 4,
+    Refunding = 5,
+    Destroyed = 6,
+}
+impl FromStr for State {
+    type Err = String;
+
+    fn from_str(input: &str) -> Result<State, Self::Err> {
+        match input {
+            "AcceptingChallenger" => Ok(State::AcceptingChallenger),
+            "MakeGuess" => Ok(State::MakeGuess),
+            "SubmitSecret" => Ok(State::SubmitSecret),
```

Comment:
Didn't implement! Moved from 2 rounds to a single round and a secret is no longer needed